### DISCUSSION TITLE
Microsoft Edge app browser does not launch AR Quick Look on iOS

### DISCUSSION THREAD

**Discussion by yosun** [google/model-viewer#4781] [Q&A] [answered]
### Description

Microsoft Edge app browser does not launch AR Quick Look on iOS

#### Live Demo
1) Try browsing to ModelViewer.dev on Edge app on iOS. 

2) The AR Quick Look functionality does not work, and instead a blank model data page arrives. 

Here is the Edge App Store link provided for reference ( I am not affiliated with them!) 
https://apps.apple.com/us/app/microsoft-edge-ai-browser/id1288723196 
### Browser Affected
 
- [ x ] Edge (iOS app) 
 

### OS
 
- [ x] iOS
 

### AR
 
- [ x] QuickLook

**Comment by elalish** [comment]
I don't believe Edge on iOS supports Quick Look, and WebXR isn't available since the only browser engine on iOS is WebKit. You're welcome to send Microsoft a feature request on it though!

**Reply by yosun** [reply]
Is there a way to have it launch quick look externally, similar to how hitting a link on iMessage launches quick look?

**Reply by elalish** [reply]
No - they're the only ones that can implement that behavior.

**Reply by yosun** [reply]
being an indie dev, codinghorror occurs when big company people push the blame to other big company people... 

since Edge is listed as supported by ModelViewer, could you guys please work with Edge iOS to get quicklook working?

**Reply by elalish** [reply]
lol, you do realize I'm basically fighting the good fight for Google open source here pretty much on my own, right? 🥲 

I don't know what Edge iOS usage percentage is, but if it's >0.1% I'm going to be really impressed (as a former Microsoft employee). There are lots of more popular browsers with no AR support (Firefox, etc), and we've been fighting the good fight with Apple to get WebXR supported on iOS so that AR can actually be halfway decent. No luck in the last 4 years, but who knows? Even at big companies, we're all just indie developers in our own way...

**Reply by yosun** [reply]
and googleplex-hearts for fighting the good fight! 

you def have way more connection resources than an indie dev here. esp since you are a former microsoft employee here - you can at least get this bug report in to hopefully the right people?

**Reply by elalish** [reply]
Thanks!

I can tell you haven't worked at Microsoft. I'll mention it to someone I know there, but I doubt they even know what building the iOS Edge engineers work in. And honestly, they pay more attention to their users (you) want than to what their competitors (me) want. I also know how features like this get prioritized (I was a PM there) and this is... not going to happen. AR never had enough users, and it doesn't have enough buzz word cache anymore either. I can barely get Apple to fix their many Quick Look regressions. 

I'm a big believer in 3D on the web, but AR... let's just say I'm a bit jaded.

**Reply by yosun** [reply]
Hang in there. XR is the inevitable next big thing. e/acc will help accelerate this timeline.

**Reply by yosun** [reply]
thanks for pushing the fix across companies ^.~

**Reply by elalish** [reply]
I'm glad I was wrong! 😄

**Comment by yosun** [comment]
Okay so since we have a menagerie of lesser known browsers with no or uncertain Model-Viewer support: - Edge, Arc etc 

For now, is it possible to have Model-Viewer display (in-lieu of a standard quick look button) a "AR Quick View Not supported on this browser - use Safari or Chrome" button? It seems to be a silent fail right now.

**Reply by milesgreen** [reply]
Edge on iOS *used* to support opening to ARQuickLook. 
After a recent update it no longer does. Seems like this was (intentionally?) removed somewhere along the way. 

Might be time to update the compatibility checks to remove Edge.

**Reply by elalish** [reply]
Yeah, we should probably update. @bghgary Do you know anything about this?

**Reply by bghgary** [reply]
I don't know anything about this. Maybe @RafaelCintron knows?

**Reply by milesgreen** [reply]
@bghgary & @RafaelCintron - if you have any insights.... I first detected Edge iOS no longer opening USDZ files around version 117. I reported it through the in-app feedback. But never heard anything.

**Comment by halton** [comment, marked as answer]
This is Halton from Microsoft Edge team, thanks for reporting this issue. In general, we should be same as Chrome and Safari, for this failure, we're seeing Edge goes into download logic, we're root causing at this issue and try to fix in manner of time.

**Reply by halton** [reply]
BTW, this issue only happen if user not signin or signin with [Microsoft Account](https://account.microsoft.com/account) like outlook/hotmail, if user signin with [Microsoft Entra ID](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id), this issue is gone. So please expect a fix.

**Reply by milesgreen** [reply]
Awesome. Great news. Thanks @halton

**Reply by halton** [reply]
Update: This issue is being fixed in Edge Canary build (only internal user can install), will cherry pick the fix to next Edge stable (on the App store version), ETA is early July.

**Reply by milesgreen** [reply]
@halton , I'm retesting this with iOS Edge 126.0.2592.86 and things are looking much better. Now able to view USDZ models in AR QuickLook. Also able to view auto generated USDZ blobs. Fixed scaling works. Custom share link also works. And everything works when opening from iFrame too. Great! Seems like this has fixed things.

**Reply by yosun** [reply]
niiice! thank you!!! ❤️ 🫶
